mods/taskbar-wheel-cycle.wh.cpp

```diff
@@ -214,6 +296,77 @@ void SwitchToTaskItem(LONG_PTR lpMMTaskListLongPtr, void* taskItem) {
     CTaskListWnd_SwitchToItem_Original(pThis_ITaskListSite, taskItem);
 }
 
+// Minimize all other windows on the same monitor as the target window
+void MinimizeOtherWindowsOnSameMonitor(HWND targetWindow) {
+    if (!targetWindow || !g_settings.minimizeOtherWindowsOnSwitch) {
+        return;
+    }
+
+    // Get the monitor of the target window
+    HMONITOR targetMonitor = MonitorFromWindow(targetWindow, MONITOR_DEFAULTTONEAREST);
+    if (!targetMonitor) {
+        Wh_Log(L"Failed to get monitor for target window");
+        return;
+    }
+
+    Wh_Log(L"Minimizing other windows on same monitor as HWND %p", targetWindow);
+
+    // Enumerate all top-level windows
+    EnumWindows(
+        [](HWND hWnd, LPARAM lParam) -> BOOL {
+            HWND targetWnd = reinterpret_cast<HWND>(lParam);
+
+            // Skip if this is the target window
+            if (hWnd == targetWnd) {
+                return TRUE;
+            }
+
+            // Skip invisible windows
+            if (!IsWindowVisible(hWnd)) {
+                return TRUE;
+            }
+
+            // Skip already minimized windows
+            if (IsIconic(hWnd)) {
+                return TRUE;
+            }
+
+            // Get window class name to filter out system windows
+            WCHAR className[256];
+            if (!GetClassName(hWnd, className, ARRAYSIZE(className))) {
+                return TRUE;
+            }
+
+            // Skip taskbar and system windows
+            if (_wcsicmp(className, L"Shell_TrayWnd") == 0 ||
+                _wcsicmp(className, L"Shell_SecondaryTrayWnd") == 0 ||
+                _wcsicmp(className, L"Progman") == 0 ||
+                _wcsicmp(className, L"WorkerW") == 0 ||
+                _wcsicmp(className, L"Windows.UI.Core.CoreWindow") == 0) {
+                return TRUE;
+            }
+
+            // Skip tool windows
+            LONG exStyle = GetWindowLong(hWnd, GWL_EXSTYLE);
+            if (exStyle & WS_EX_TOOLWINDOW) {
+                return TRUE;
+            }
+
+            // Check if window is on the same monitor
+            HMONITOR windowMonitor = MonitorFromWindow(hWnd, MONITOR_DEFAULTTONEAREST);
+            HMONITOR targetMonitor = MonitorFromWindow(targetWnd, MONITOR_DEFAULTTONEAREST);
+
+            if (windowMonitor == targetMonitor) {
+                // Minimize this window
+                ShowWindow(hWnd, SW_MINIMIZE);
```

Comment:
This won't work for elevated windows (running as admin). This can be used instead:
```cpp
    if (CanMinimizeWindow(hWnd)) {
        DWORD dwProcessId;
        GetWindowThreadProcessId(hWnd, &dwProcessId);
        AllowSetForegroundWindow(dwProcessId);
        return PostMessage(hWnd, WM_SYSCOMMAND, SC_MINIMIZE, 0);
    }
```

`CanMinimizeWindow` can be found [here](https://github.com/m417z/my-windhawk-mods/blob/5fe290cba141168c8412f2f818152acacb5fee1d/mods/taskbar-button-scroll.wh.cpp#L387).

The `AllowSetForegroundWindow` part can help avoid flashing taskbar buttons.